### PR TITLE
Update python simulation class

### PR DESCRIPTION
**Pull request by tanishqaggarwal**
This is the set of the latest changes required on the Python side for FSW-psim integration.

### CODE REVIEW

File: python/usb_console/simulation.py [hunk 1]
```diff
@@ -87,28 +86,34 @@ def run(self):
...
             self.sim_time = self.main_state['follower']['dynamics']['time']
 
-            # Get sensor readings
+            # Step 1. Get sensor readings from simulation
             self.sensor_readings_follower = self.eng.sensor_reading(self.main_state['follower'],self.main_state['leader'], nargout=1)
             self.sensor_readings_leader = self.eng.sensor_reading(self.main_state['leader'],self.main_state['follower'], nargout=1)
-            # update dynamics
+            
+            # Step 2. Update dynamics
             self.main_state = self.eng.main_state_update(self.main_state, nargout=1)
```
Comment (tanishqaggarwal) [inline python/usb_console/simulation.py]: @nhz2 How do I get this to run in the background?

File: python/usb_console/simulation.py [hunk 2]
```diff
@@ -87,28 +86,34 @@ def run(self):
...
-            # simulate flight computers
+            
+            # Step 3. Simulate flight computers
+            # Step 3.1. Use MATLAB simulation as a base
```
Comment (tanishqaggarwal) [inline python/usb_console/simulation.py:97]: By "base" I mean that some of the actuator commands come from the simulation itself, when the Flight Controller hasn't yet implemented a certain control. Gradually, calling update_FC_state will become unnecessary here once the FlightController is fully autonomous.

File: python/usb_console/simulation.py [hunk 3]
```diff
@@ -87,28 +86,34 @@ def run(self):
...
             self.computer_state_follower, self.actuator_commands_follower = \
                 self.eng.update_FC_state(self.computer_state_follower,self.sensor_readings_follower, nargout=2)
             self.computer_state_leader, self.actuator_commands_leader = \
                 self.eng.update_FC_state(self.computer_state_leader,self.sensor_readings_leader, nargout=2)
-            # command actuators
+            
+            # Step 3.2. Send inputs and read outputs from Flight Computer
```
Comment (tanishqaggarwal) [inline python/usb_console/simulation.py:103]: Currently blank until @shihaocao can get an estimator task rolling.

File: python/usb_console/simulation.py [hunk 4]
```diff
@@ -87,28 +86,34 @@ def run(self):
...
+
+            # Step 3.3. Tell flight computer to run its cycle
+            self.flight_controller_leader.write_state("cycle.start", "true")
+            self.flight_controller_follower.write_state("cycle.start", "true")
```
Comment (tanishqaggarwal) [inline python/usb_console/simulation.py:107]: This is a new feature. Only if `cycle.start` is true does the flight software continue on to the "actuate" portion of its control cycle.

File: python/usb_console/simulation.py (hunk 1 above)
Comment (kylekrol) [reply python/usb_console/simulation.py]: @tanishqaggarwal It's a futures type thing. See the documentation here:
https://www.mathworks.com/help/matlab/matlab_external/call-matlab-functions-asynchronously-from-python.html

**Review by kylekrol — DISMISSED** [review]
Approving and leaving it up to you whether you want to make the MATLAB calls asynchronous or not. It may not be necessary right away.

File: python/usb_console/simulation.py (hunk 1 above)
Comment (tanishqaggarwal) [reply python/usb_console/simulation.py]: Thanks!

**Comment by nhz2** [comment]
I'm having trouble running the sim,

```
(venv) (base) Nathans-MBP:python nathan$ python usb_console/run_simulation.py -c usb_console/configs/fc_only_native.json

Traceback (most recent call last):

  File "usb_console/run_simulation.py", line 212, in <module>
    with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), 'configs/radio_keys.json')) as radio_keys_config_file:

FileNotFoundError: [Errno 2] No such file or directory: '/Users/nathan/Documents/GitHub/psim/python/usb_console/configs/radio_keys.json'
```

**Comment by tanishqaggarwal** [comment]
@nhz2 Ask Fatima for the radio keys file.

**Comment by kylekrol** [comment]
What exactly keeps happening with gtest? That isn't a submodule anymore.

**Review by kylekrol — APPROVED** [review]

File: python/usb_console/simulation.py (hunk 2 above)
Comment (nhz2) [reply python/usb_console/simulation.py:97]: you can also use `function actuator_commands = actuators_off_command()` to generate all the commands that will turn off the actuators as a base.

**Pull request merged**